### ISSUE TITLE
I have a question about the requirements for SAM2 fine-tuning.

### ISSUE THREAD

**Issue by smg0411** [facebookresearch/sam2#606]
First of all, I appreciate your outstanding achievements. I am planning to fine-tune SAM2 with my custom data. I have read the documentation on fine-tuning, but I couldn't find any information regarding prompt inputs. I was wondering if SAM2 does not require specific input prompts during the fine-tuning process. I would appreciate it if you could clarify this. Thank you.

**Comment by ckapelonis02** [comment]
You might want to take a look at [this](https://github.com/sagieppel/fine-tune-train_segment_anything_2_in_60_lines_of_code), if you haven't already.

**Comment by sun913** [comment]
This is a good question. SAM2 will use mask, point, or bounding box as input prompts during the fine-tuning process. The mask prompt will use the ground truth mask as the input, and the points or bounding boxes will be generated by using the ground truth mask during the process. If you check /sam2-main/training/model/sam2.py file, which is the model class file used in fine-tuning. You can see

![Image](https://github.com/user-attachments/assets/c6d4cddb-717c-40a4-bd03-213342e7fdcc)
use_pt_input means whether not to use mask prompt (from my understanding) with a probability of 0.5. prob_to_use_box_input means whether to use the box prompt with a probability of P(box) = prob_to_use_pt_input * prob_to_use_box_input, which is 0.25. So the probability of using point prompt is 0.25 as well. 

Hope this answers your question.
